src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2754,8 +2755,12 @@ public void runMayThrow()
                     else
                         logger.debug("Discarding hint for endpoint not part of ring: {}", target);
                 }
-                logger.trace("Adding hints for {}", validTargets);
-                HintsService.instance.write(hostIds, Hint.create(mutation, currentTimeMillis()));
+                long creationTime = currentTimeMillis();
+                if (DatabaseDescriptor.isHintTtlUseMutationCreationTime()) {
+                    creationTime -= TimeUnit.MILLISECONDS.convert(Math.abs(approxTime.now() - mutation.getApproxCreatedAtNanos()), NANOSECONDS);
```

Comment:
would it make more sense to use the time from the PartitionUpdates DeletionTime inside the mutation instead of the time the mutation is created on the coordinator? I could see where that would be iffy since theres no guarantees client isn't using their own thing for writetimes.